articles/speakeasy-two-factor-authentication-in-nodejs/index.md

```diff
@@ -0,0 +1,245 @@
+First we will set up our server. The scripts below will will help us set up our server. Our server will run on the application's entry point `index.js` file. So add the below block of code in the index file.
+```js
+const express = require('express')
+const app = express();
+app.use(express.json())
+const PORT = process.env.PORT || 5000
+
+```
+After setting the server, we bring our dependencies into the `index.js` file where the server script runs. The `uuid` will create a unique user id, while the `node-json-db` will serve as the database to store the user-id and the secret associated with that user-id.
+```js
+//adding the speakeasy library
+const speakeasy = require('speakeasy')
+
+//adding the uuid dependacy
+const uuid = require('uuid')
+
+//adding node-json-db
+const { JsonDB } = require('node-json-db')
+const { Config } = require('node-json-db/dist/lib/JsonDBConfig')
+```
+
+### The node json databse
+Our application will use the node json database to store user records in json format. Check out  [this link](https://www.npmjs.com/package/node-json-db?S_TACT=AGRAK616). To find more about `node-json-db`. From the documentation of `node-json-db`, to initialize the `node-json-db`, we will add the script below.
+```js
+
+// instance of the node-json-db
+const db = new JsonDB(new Config("DataBase", true, false, '/'));
+```
+ * `new Config`  - Creates a node-json-db database config
+ * `DataBase` - specifies the name of the JSON storage file
+ * `true` Tells the database to save on each push.
+ * `false` Instructs the database to save the database in a human-readable format
+ * `/` - the separator to use when accessing  our database values
+
+
+
+### Sending requests using postman
+Since our application does not have a frontend, we will use postman for sending the requests to the applications' backend. Postman provides an interface for handling requests that would have otherwise been handled by the HTML. Check out [this guide](https://www.blazemeter.com/blog/how-use-postman-manage-and-execute-your-apis) on how to use postman. You can download postman using [this](https://www.postman.com/downloads/) link. 
+In postman, we will use three routes, register, verify and validate route. So we will create the URLs as below:
+
+- Register: http://localhost:5000/api/register
+- Verify: http://localhost:5000/api/verify
+- Validate: http://localhost:5000/api/validate
+
+### Registering users
+In this application, we assume that a user is authetcated using his/her user-id. We therefore ignore other user details. We will resgister users and store their `user-id`  alongside the `secret-key` generated by speakeasy in the `Database.json` file. The registration process  begins by passing a post request to the `register` route in the `index.js` file. We then use `uuid` to generate a unique `user-id` and generate a temporary `secret-key` for the user ID. Next, we store the `user-id` and the `secret-key` in the `node-json-db`. The code for the process is as below:
```

Comment:
```suggestion
In this application, we assume that a user is authenticated using his/her user-id. We therefore ignore other user details. We will register users and store their `user-id`  alongside the `secret-key` generated by speakeasy in the `Database.json` file.

The registration process begins by passing a post request to the `register` route in the `index.js` file. We then use `uuid` to generate a unique `user-id` and generate a temporary `secret-key` for the user ID.

Next, we store the `user-id` and the `secret-key` in the `node-json-db`.

The code for the process is as below:
```

Reply:
Always break down bigger paragraphs into smaller ones to enhance the readability.

One paragraph must speak only about one idea